From 8f7c2e9d0fe19140b8b1774dd67c80a8f5d59488 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@freescale.com>
Date: Mon, 9 Feb 2015 18:19:38 +0800
Subject: [PATCH] MLK-10214-2: regmap: rbtree: Fix the shift issue.

The right shift should be changed to left shift. Lower register
address correspond to LSB of the cache_present, if right shift
old cache_present, the valid bits is moved out, then the new
cache_present is not correct.

Signed-off-by: Shengjiu Wang <shengjiu.wang@freescale.com>
---
 drivers/base/regmap/regcache-rbtree.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/base/regmap/regcache-rbtree.c b/drivers/base/regmap/regcache-rbtree.c
index 930cad4..2b946bc 100644
--- a/drivers/base/regmap/regcache-rbtree.c
+++ b/drivers/base/regmap/regcache-rbtree.c
@@ -313,7 +313,7 @@ static int regcache_rbtree_insert_to_block(struct regmap *map,
 	if (pos == 0) {
 		memmove(blk + offset * map->cache_word_size,
 			blk, rbnode->blklen * map->cache_word_size);
-		bitmap_shift_right(present, present, offset, blklen);
+		bitmap_shift_left(present, present, offset, blklen);
 	}
 
 	/* update the rbnode block, its size and the base register */
-- 
1.8.0

